bakery/src/tasks/upload-docx.js

```diff
@@ -0,0 +1,53 @@
+const dedent = require('dedent')
+
+const { constructImageSource } = require('../task-util/task-util')
+
+const task = (taskArgs) => {
+  const { parentGoogleFolderId, awsAccessKeyId, awsSecretAccessKey, queueStateBucket, codeVersion } = taskArgs
+  const imageDefault = {
+    name: 'openstax/cops-bakery-scripts',
+    tag: 'trunk'
+  }
+  const imageOverrides = taskArgs != null && taskArgs.image != null ? taskArgs.image : {}
+  const imageSource = constructImageSource({ ...imageDefault, ...imageOverrides })
+
+  return {
+    task: 'upload docx',
+    config: {
+      platform: 'linux',
+      image_resource: {
+        type: 'docker-image',
+        source: imageSource
+      },
+      params: {
+        GOOGLE_SERVICE_ACCOUNT_CREDENTIALS: '((google-service-account-credentials))',
+        AWS_ACCESS_KEY_ID: `${awsAccessKeyId}`,
+        AWS_SECRET_ACCESS_KEY: `${awsSecretAccessKey}`
+      },
+      inputs: [
+        { name: 'book' },
+        { name: 'docx-book' }
+      ],
+      run: {
+        path: '/bin/bash',
+        args: [
+          '-ce', // FIXME: Not including -x to avoid displaying credentials when writing to file.
+          dedent`
+          echo "$GOOGLE_SERVICE_ACCOUNT_CREDENTIALS" > /tmp/service_account_credentials.json
+          collection_id="$(cat book/collection_id)"
+          book_legacy_version="$(cat book/version)"
+          docx_dir="docx-book/$collection_id/docx"
+          book_metadata="docx-book/$collection_id/raw/metadata.json"
+          book_title="$(cat $book_metadata | jq -r '.title')"
+          upload-docx "$docx_dir" "$book_title" "${parentGoogleFolderId}" /tmp/service_account_credentials.json
+          complete_filename=".$collection_id@$book_legacy_version.complete"
+          date -Iseconds > "/tmp/$complete_filename"
+          aws s3 cp "/tmp/$complete_filename" "s3://${queueStateBucket}/${codeVersion}/$complete_filename"
```

Comment:
Do gdoc pipeline and web hosting pipeline share the same queue state file names per version? If we spin up each pipeline with the same version, what prevents them from affecting each other's visible states? I'm just now realizing the check-feed job that I pushed up may share this issue :fearful:

Reply:
Good catch (I can update the `check-feed` side in tandem with this). We should probably tease apart the `QUEUE_FILENAME` and `FEED_FILE_URL` values for `gdoc` too, no? (so they use the same `S3_QUEUE_STATE_BUCKET`, but we can add `GDOC_QUEUE_FILENAME` and `GDOD_FEED_FILE_URL`).

Reply:
I added a prefix to the state files so we can keep things separate. Just to be sure, I tested by running concurrent pipelines of dist and gdoc that used the same tag, and they didn't trample each other.